File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        if(nums == null || nums.length == 0) return new int[]{-1, -1};
        //find first element in sorted array
        int l = 0, r = nums.length - 1;
        while(l < r) {
            int m = (l + r) / 2;
            if(nums[m] >= target) {
                r = m;
            } else {
                l = m + 1;
            }
        }
        if(nums[l] != target) return new int[]{-1, -1};
        int left = l;
        r = nums.length - 1;
        
        while( l < r) {
            int m = (l + r + 1) / 2;
            if(nums[m] <= target)
                l = m;
            else 
                r = m - 1;
        }
        
        int right = l;
        return new int[]{left, right};
    }
}

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        res = [-1, -1]
        if not nums: return res
        l, r=  0, len(nums) -1
        while l < r:
            m = (l + r) // 2
            if nums[m] >= target:
                r = m
            else:
                l = m + 1
        if nums[l] != target:
            return [-1, -1]
        res[0] = l
        r = len(nums) - 1
        while l < r:
            m = (l + r + 1) // 2
            if nums[m] <= target:
                l = m
            else:
                r = m - 1
        res[1] = l
        return res